packages/babel-plugin-jsx/src/utils.ts

```diff
@@ -50,6 +50,10 @@ export const checkIsComponent = (path: NodePath<t.JSXOpeningElement>): boolean =
 
   const tag = (namePath as NodePath<t.JSXIdentifier>).node.name;
 
+  if (state.opts.isCustomElement && state.opts.isCustomElement(tag)) {
+    return false;
+  }
+
   return shouldTransformedToSlots(tag) && !htmlTags.includes(tag) && !svgTags.includes(tag);
```

Comment:
```suggestion
  return  state.opts.isCustomElement?.(tag) && shouldTransformedToSlots(tag) && !htmlTags.includes(tag) && !svgTags.includes(tag);
```

Reply:
state.opts.isCustomElement?.(tag) => Type 'boolean | undefined' is not assignable to type 'boolean'

Reply:
sorry
You can try `!!state.opts.isCustomElement?.(tag)`.

Reply:
!!!(state.opts.isCustomElement?.(tag)) => error  Redundant double negation  no-extra-boolean-cast
add /* eslint-disable no-extra-boolean-cast */ ?

Reply:
`!!` not `!!!`

Reply:
checkIsComponent should be false when isCustomElement is true